language-plutus-core/plutus-ir-test/Spec.hs

```diff
@@ -61,8 +56,12 @@ compileAndMaybeTypecheck
     -> Term TyName Name uni a
     -> Except (Error uni (Provenance a)) (PLC.Term TyName Name uni (Provenance a))
 compileAndMaybeTypecheck doTypecheck pir = flip runReaderT defaultCompilationCtx $ runQuoteT $ do
+    -- FIXME: have a doTypecheck knob  for `PIR.compileTerm` as well
     compiled <- compileTerm pir
-    when doTypecheck $ void $ PLC.inferType PLC.defConfig compiled
+    when doTypecheck $ do
+        -- trick for prism-error lifting
+        plcConcrete <- runExceptT $ void $ PLC.inferType PLC.defConfig compiled
+        liftEither $ first (view (re _PLCError)) plcConcrete
```

Comment:
works, thanks @effectfully

Reply:
It works in this place because the error is concretely declared in the typesig to be (PLC.Error). It does not work in other places where the Error is left abstract, unfortunately. That's why I used this `re` thing.